go/consensus/tendermint/light/service.go

```diff
@@ -0,0 +1,337 @@
+func (*client) Name() string {
+	return "tendermint/light"
+}
+
+func (c *client) Start() error {
+	go c.worker()
+
+	return nil
+}
+
+func (c *client) GetStatus(ctx context.Context) (*consensus.LightClientStatus, error) {
+	status := &consensus.LightClientStatus{}
+	oldest, err := c.store.FirstLightBlockHeight()
+	if err != nil {
+		return nil, err
+	}
+	status.OldestHeight = oldest
+	if oldest > -1 { // -1 indicates empty store.
+		var lb *tmtypes.LightBlock
+		lb, err = c.store.LightBlock(oldest)
+		if err != nil {
+			return nil, err
+		}
+		status.OldestHash = hash.LoadFromHexBytes(lb.Header.Hash())
+		status.OldestTime = lb.Time
+	}
+
+	latest, err := c.store.LastLightBlockHeight()
+	if err != nil {
+		return nil, err
+	}
+	status.LatestHeight = latest
+	if latest > -1 { // -1 indicates empty store.
+		var lb *tmtypes.LightBlock
+		lb, err := c.store.LightBlock(latest)
+		if err != nil {
+			return nil, err
+		}
+		status.LatestHash = hash.LoadFromHexBytes(lb.Header.Hash())
+		status.LatestTime = lb.Time
+	}
+
+	for _, p := range c.providers {
+		status.PeerIDs = append(status.PeerIDs, p.PeerID())
+	}
+
+	return status, nil
+}
+
+func (c *client) Stop() {
+	if !atomic.CompareAndSwapUint32(&c.stopped, 0, 1) {
+		return
+	}
+	close(c.stopCh)
+}
+
+func (c *client) Quit() <-chan struct{} {
+	return c.quitCh
+}
+
+func (c *client) Cleanup() {
+}
+
+func (c *client) consensusBlkToTendermint(lb *consensus.LightBlock) (*tmtypes.LightBlock, error) {
+	// Convert to protobuff LightBlock.
+	var pb tmproto.LightBlock
+	if err := pb.Unmarshal(lb.Meta); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal tendermint protobuff light block: %w", err)
+	}
+	// Convert to tendermint LightBlock.
+	tmLb, err := tmtypes.LightBlockFromProto(&pb)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert protobuff light block into tendermint light block: %w", err)
+	}
+	return tmLb, nil
+}
+
+func (c *client) worker() {
+	if !c.enabled {
+		return
+	}
+	var initOk bool
+	defer func() {
+		if !initOk {
+			close(c.quitCh)
+		}
+	}()
+
+	// Wait for consensus to be synced.
+	select {
+	case <-c.stopCh:
+		return
+	case <-c.ctx.Done():
+		return
+	case <-c.consensus.Synced():
+	}
+
+	// Store latest light block into the trusted store.
+	lb, err := c.consensus.GetLightBlock(c.ctx, consensusAPI.HeightLatest)
+	if err != nil {
+		c.logger.Error("failed to obtain latest light block from consensus", "err", err)
+		return
+	}
+	tmLb, err := c.consensusBlkToTendermint(lb)
+	if err != nil {
+		c.logger.Error("failed to convert consensus light block to tendermint", "err", err)
+		return
+	}
+	if err = c.store.SaveLightBlock(tmLb); err != nil {
+		c.logger.Error("failed to initialize light client store with a trusted block", "err", err)
+		return
+	}
+
+	chainCtx, err := c.consensus.GetChainContext(c.ctx)
+	if err != nil {
+		c.logger.Error("failed to obtain chain context", "err", err)
+		return
+	}
+	tmGenDoc, err := tmapi.GetTendermintGenesisDocument(c.genesis)
```

Comment:
This seems expensive just to get the Tendermint's chain context. Maybe expose something like `tmapi.TendermintChainID(chainContext string)` and call that instead on `chainCtx`?